src/pydata_sphinx_theme/translator.py

```diff
@@ -23,11 +24,35 @@ def __init__(self, *args, **kwds):
         self.settings.table_style = "table"
 
     def starttag(self, *args, **kwargs):
-        """Ensure an aria-level is set for any heading role."""
+        """Perform small modifications to tags.
+
+        - ensure aria-level is set for any tag with heading role
+        - ensure <pre> tags have tabindex="0".
+        """
         if kwargs.get("ROLE") == "heading" and "ARIA-LEVEL" not in kwargs:
             kwargs["ARIA-LEVEL"] = "2"
+
+        if "pre" in args:
+            kwargs["tabindex"] = "0"
```

Comment:
If only this worked for everything. But Sphinx gives literal blocks special treatment, and most of the various different kinds of code blocks and other blocks that result in a `<pre>` tag are some kind of literal block.